### ISSUE TITLE
Two naming/consistency follow-ups from #519

### ISSUE THREAD

**Issue by louisrf** [ocean-data-factory-sweden/kso#520]
1. In `src/kso/serving_utils.py`, line 475: #519 csv export writes inference detections to `annotations.csv`, I'd like to have `detections.csv` to better distinguish model inference output from annotation data (Biigle CSVs, YOLO labels).

2. I think we should use the same MODEL_NAME in the YAML, MLflow, and the Ultralytics run folder (runs/detect/<MODEL_NAME>)

Currently `TrainingManager.train_model()` does not pass a `name` argument to Ultralytics, so training runs land in `runs/detect/train/`, `train2/`, `train3/`, etc. The MLflow registered model name is decoupled from the 
Ultralytics run folder, which makes it hard to trace which folder produced which weights.

In `src/kso/trainer.py`, if you could add `name=model_name` to the `model.train()` call inside `train_model()`:

Then it will match the MLflow registry entry. If a folder with that name already exists, Ultralytics raises (which is the behaviour we want per the team convention: new run = new name).

Practical example i have in mind:
PROJECT_NAME = "ifremer_2026"
MODEL_NAME = "ifremer_yolo11l_50ep" # first training attempt
MODEL_NAME = "ifremer_yolo11xl_100ep" # second training attempt
MODEL_NAME = "ifremer_yolo26m_finetune_100ep"

**Comment by GhaithChaabane** [comment]
Passed model_name to model.train(name=...) so ultralytics run directories match mlflow registered model names and
changed inference detections to detections.csv instead of annotations.csv.
this will be added in the next PR.
